dali/kernels/mask/grid_mask_cpu.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_MASK_GRID_MASK_CPU_H_
+#define DALI_KERNELS_MASK_GRID_MASK_CPU_H_
+
+#include <utility>
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+
+template<typename Type>
+class GridMaskCpu {
+ public:
+  KernelRequirements Setup(KernelContext &context, const TensorShape<> &shape) {
+    KernelRequirements req;
+    req.output_shapes = { TensorListShape<>{{shape}} };
+    return req;
+  }
+
+  void Run(KernelContext &context, const OutTensorCPU<Type> &out,
+           const InTensorCPU<Type> &in, int tile, float ratio, float angle) {
+    auto in_ptr = in.data;
+    auto out_ptr = out.data;
+    float dx = cos(angle) / tile;
+    float dy = sin(angle) / tile;
+
+    for (int y = 0; y < in.shape[0]; y++) {
+      float fx = y * -sin(angle) / tile;
```

Comment:
You could extract the calculation of `-sin(angle) / tile` and `cos(angle) / tile` outside of the loop for better performance.

Reply:
But this may affect the accuracy.